packages/node/src/main.ts

```diff
@@ -11,7 +12,18 @@ const logger = getLogger('subql-node');
 
 async function bootstrap() {
   const debug = argv('debug');
-  const port = argv('port') as number;
+  const candidatePort = argv('port') as number;
+  const port = await findAvailablePort(candidatePort);
```

Comment:
I think it would be more expected if we only do port scan when --port is not given (default port is used).

Reply:
moved default port outside of yargs so that we can detect if user is using flag